lang/docs/src/doc.rs

```diff
@@ -1,127 +1,52 @@
 use std::fs;
-use std::io;
 use std::io::prelude::*;
-use std::path::Path;
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
 
+use askama::Template;
+use html_escape::decode_html_entities;
 use opener;
 
-use driver::paths::CSS_PATH;
+use driver::paths::{CSS_PATH, CSS_TEMPLATE_PATH};
 use driver::Database;
 use printer::{Print, PrintCfg};
 
-const HTML_END: &str = " </code></pre>
-    </div></body></html>";
-
-fn html_start(filepath: &Path) -> String {
-    format!(
-        "<!DOCTYPE html>
-<html lang=\"en\">
-<head>
-    <meta charset=\"UTF-8\">
-    <meta name=\"viewport\" content=\"width=device-width, initial-scale=1.0\">
-    <title>{filename}</title>
-    <link rel=\"stylesheet\" href=\"style.css\">
-</head>
-<body>
-<div>
-        <h1>{filename}</h1>
-        <pre><code>",
-        filename = filepath.file_name().unwrap().to_string_lossy()
-    )
-}
-
 pub async fn write_html(filepath: &PathBuf, htmlpath: &PathBuf) {
     let mut db = Database::from_path(filepath);
     let uri = db.resolve_path(filepath).expect("Failed to resolve path");
     let prg = db.ust(&uri).await.expect("Failed to get UST");
     let cfg = PrintCfg::default();
 
```

Comment:
At this point, if the `target_pol/docs` directory doesn't exist, create it (recursively, e.g. using [`fs::create_dir_all`](https://doc.rust-lang.org/stable/std/fs/fn.create_dir_all.html)). Currently, the code produces the following error when `target_pol` is missing;

```
Error:   × Main thread panicked.
  ├─▶ at /home/tim/Dev/polarity-lang/polarity/lang/docs/src/doc.rs:22:51
  ╰─▶ Failed to copy CSS file: Os { code: 2, kind: NotFound, message: "No such
      file or directory" }
  help: set the `RUST_BACKTRACE=1` environment variable to display a
        backtrace.
```